src/endpoint.jl

```diff
@@ -80,26 +80,59 @@ function Base.isequal(a::RightEndpoint, b::LeftEndpoint)
 end
 
 function Base.isless(a::LeftEndpoint, b::LeftEndpoint)
-    a.endpoint < b.endpoint || (a.endpoint == b.endpoint && a.included && !b.included)
+    if isnothing(a.endpoint) && isnothing(b.endpoint)
```

Comment:
all of the `isnothing(x)`s should `x === nothing` for reasons i don't pretend to understand but which is referenced in https://docs.julialang.org/en/v1/manual/performance-tips/#Checking-for-equality-with-a-singleton-1 (xref https://github.com/JuliaLang/julia/issues/35585)

Reply:
That's true. We may want to have a `isunbounded` abstraction as we may switch our internal representation

Reply:
ah interesting. Will do. I had read somewhere to use `isnothing(x)` but that makes sense

Reply:
Actually, I believe `isnothing(x)` should be as performant as `x === nothing` it's `x == nothing` that can be slightly slower

Reply:
I like the abstraction idea. I'll throw that in as well